fbchat/client.py

```diff
@@ -794,71 +771,120 @@ def _parseMessage(self, content):
+                            recipient = msg_metadata['threadKey']['otherUserFbId']
+                            message_type = MsgType.USER
+                        return recipient, message_type
+
+                    delta = m["delta"]
+                    delta_type = delta.get("type")
+                    metadata = delta.get("messageMetadata")
+
+                    # Added participants
+                    if 'addedParticipants' in delta:
+                        added_ids = [x['userFbId'] for x in delta['addedParticipants']]
+                        author_id = metadata['actorFbId']
+                        thread_id = metadata['threadKey']['threadFbId']
+                        ts = metadata["timestamp"]
+                        self.onPeopleAdded(added_ids=added_ids, author_id=author_id, thread_id=thread_id, ts=ts)
+
+                    # Left/removed participants
+                    elif 'leftParticipantFbId' in delta:
+                        removed_id = delta['leftParticipantFbId']
+                        author_id = metadata['actorFbId']
+                        thread_id = metadata['threadKey']['threadFbId']
+                        ts = metadata["timestamp"]
+                        self.onPersonRemoved(removed_id=removed_id, author_id=author_id, thread_id=thread_id, ts=ts)
+
+                    # Seen
+                    elif delta.get("class") == "ReadReceipt":
+                        seen_by = delta["actorFbId"] or delta["threadKey"]["otherUserFbId"]
+                        thread_id = delta["threadKey"].get("threadFbId")
+                        timestamp = delta["actionTimestampMs"]
+                        self.onSeen(seen_by=seen_by, thread_id=thread_id, timestamp=timestamp)
+
+                    # Color change
+                    elif delta_type == "change_thread_theme":
+                        mid = metadata["messageId"]
+                        author_id = metadata["actorFbId"]
+                        new_color = delta["untypedData"]["theme_color"]
+                        changed_for, msg_type = getRecipientIdAndMsgType(metadata)
+                        ts = metadata["timestamp"]
+                        self.onColorChange(mid=mid, author_id=author_id, new_color=new_color, changed_for=changed_for,
+                                           msg_type=msg_type, ts=ts)
+
+                    # Title change
+                    elif delta_type == "change_thread_nickname":
+                        mid = metadata["messageId"]
+                        author_id = metadata["actorFbId"]
+                        changed_for = delta["untypedData"]["participant_id"]
+                        new_title = delta["untypedData"]["nickanme"]
+                        thread_id = metadata["threadKey"].get("threadFbId")
+                        ts = metadata["timestamp"]
+                        self.onTitleChange(mid=mid, author_id=author_id, changed_for=changed_for, new_title=new_title,
```

Comment:
Yeah, I agree, it's possible, but it's rare that when implementing some sort of bot, that you'd want to act differently on groups than on private chats. That's what I mean by duplicate code, not so much duplicate code in `fbchat`, but duplicate code in user code. Apart from that, functions like `send`, `sendRemoteImage` and so on should also take a `MsgType` as a parameter instead of `is_user`